purchase_last_price_info/models/product.py

```diff
@@ -1,29 +1,35 @@
 # -*- encoding: utf-8 -*-
-##############################################################################
-# For copyright and license notices, see __openerp__.py file in root directory
-##############################################################################
 
-from openerp import api, fields, models
+from odoo import api, fields, models
+import odoo.addons.decimal_precision as dp
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
-    @api.one
+    @api.multi
     def _get_last_purchase(self):
         """ Get last purchase price, last purchase date and last supplier """
-        lines = self.env['purchase.order.line'].search(
-            [('product_id', '=', self.id),
-             ('state', 'in', ['confirmed', 'done'])]).sorted(
-            key=lambda l: l.order_id.date_order, reverse=True)
-        self.last_purchase_date = lines[:1].order_id.date_order
-        self.last_purchase_price = lines[:1].price_unit
-        self.last_supplier_id = lines[:1].order_id.partner_id
+        for product in self:
+            line = self.env['purchase.order.line'].search([
+                ('product_id', '=', self.id),
+                ('state', 'in', ['purchase', 'done'])
+            ], order='date_order DESC', limit=1)
+            product.last_purchase_date = line.date_order
+            product.last_purchase_price = line.price_unit
+            product.last_supplier_id = line.partner_id
 
     last_purchase_price = fields.Float(
-        string='Last Purchase Price', compute='_get_last_purchase')
+        string='Last Purchase Price',
+        compute='_get_last_purchase',
```

Comment:
@Abranes IMO method name should be `_compute_last_purchase`.

same other

Reply:
@nikul-serpentcs Can you update your review?